pkg/alertmanager/multitenant.go

```diff
@@ -401,7 +402,25 @@ func (am *MultitenantAlertmanager) transformConfig(userID string, amConfig *amco
 		}
 	}
 
-	return &newConfig, nil
+	return newConfig, nil
+}
+
+func copyConfig(config *amconfig.Config) (*amconfig.Config, error) {
```

Comment:
maybe add a comment explaining we need to do a deep copy because of pointers, slices, etc